pype/plugins/global/publish/extract_review.py

```diff
@@ -13,22 +19,1064 @@ class ExtractReview(pyblish.api.InstancePlugin):
+        input_height = input_data["height"]
+
+        self.log.debug("pixel_aspect: `{}`".format(pixel_aspect))
+        self.log.debug("input_width: `{}`".format(input_width))
+        self.log.debug("resolution_height: `{}`".format(input_height))
+
+        # Skip processing if both conditions are not met
+        if "reformat" not in new_repre["tags"] and not letter_box:
+            new_repre["resolutionWidth"] = input_width
+            new_repre["resolutionHeight"] = input_height
+            return filters
+
+        # NOTE Setting only one of `width` or `heigth` is not allowed
+        output_width = output_def.get("width")
+        output_height = output_def.get("height")
+        # Use instance resolution if output definition has not set it.
+        if output_width is None or output_height is None:
+            output_width = temp_data["resolution_width"]
+            output_height = temp_data["resolution_height"]
+
+        # Use source's input resolution instance does not have set it.
+        if output_width is None or output_height is None:
+            output_width = input_width
+            output_height = input_height
+
+        # defining image ratios
+        input_res_ratio = (
+            (float(input_width) * pixel_aspect) / input_height
+        )
+        output_res_ratio = float(output_width) / float(output_height)
+        self.log.debug("resolution_ratio: `{}`".format(input_res_ratio))
+        self.log.debug("output_res_ratio: `{}`".format(output_res_ratio))
+
+        # Round ratios to 2 decimal places for comparing
+        input_res_ratio = round(input_res_ratio, 2)
+        output_res_ratio = round(output_res_ratio, 2)
+
+        # get scale factor
+        scale_factor_by_width = (
+            float(output_width) / (input_width * pixel_aspect)
+        )
+        scale_factor_by_height = (
+            float(output_height) / (input_height * pixel_aspect)
+        )
+
+        self.log.debug(
+            "scale_factor_by_with: `{}`".format(scale_factor_by_width)
+        )
+        self.log.debug(
+            "scale_factor_by_height: `{}`".format(scale_factor_by_height)
+        )
+
+        # letter_box
+        letter_box = output_def.get("letter_box")
+        if letter_box:
+            ffmpeg_width = output_width
+            ffmpeg_height = output_height
+            if "reformat" in new_repre["tags"]:
+                if input_res_ratio == output_res_ratio:
+                    letter_box /= pixel_aspect
+                elif input_res_ratio < output_res_ratio:
+                    letter_box /= scale_factor_by_width
+                else:
+                    letter_box /= scale_factor_by_height
+            else:
+                letter_box /= pixel_aspect
+                if input_res_ratio != output_res_ratio:
+                    ffmpeg_width = input_width
+                    ffmpeg_height = int(input_height * pixel_aspect)
+
+            # QUESTION Is scale required when ffmpeg_width is same as
```

Comment:
We're always using this scale filter but I think it is not required...

Reply:
Well the question is what should trigger it then. It is needed in cases where pixel_aspect or reformat is specified at the very least. Or have we dropped `reformat` tag and simply reformat if target resolution is set now?

Reply:
perhaps as we are using `iw` and `ih` in `drawbox=0:0:iw:round((ih-(iw...` we should but it is just a guess

Reply:
It is required for proper letter box scaling when aspect ratio of input is different then output's.